### PR TITLE
Add SBL setup support for QEMU

### PR DESCRIPTION
**Pull request by mauricema**
This patch set added basic infrastructure to support SBL setup through MicroPython payload module.

### CODE REVIEW

File: PayloadPkg/OsLoader/OsLoader.c [hunk 1]
```diff
@@ -668,13 +673,36 @@ StartBooting (
...
+          // So build a path like !SETP/???? for command line.
+          CopyMem (ModCmdLineBuf, PathPtr, 6);
+          CopyMem (ModCmdLineBuf + 6, PathPtr + 11, 4);
+          ModCmdLineBuf[15] = 0;
+        } else {
+          ModCmdLineBuf[0] = 0;
+        }
+        PldModParam.CmdLineBuf = ModCmdLineBuf;
+      }
+    }
```
Comment (gdong1) [inline PayloadPkg/OsLoader/OsLoader.c:697]: Had better these code is built out by FeaturePcdGet (PcdEnableSetup) for size.

File: PayloadPkg/OsLoader/OsLoader.c [hunk 2]
```diff
@@ -1219,11 +1249,15 @@ PayloadMain (
...
-      BootShell    = TRUE;
+    if (OsBootOptionList->RestrictedBoot != 0) {
+      BootShell    = FALSE;
+    } else {
+      if (OsBootOptionList->BootToShell != 0) {
+        BootShell    = TRUE;
+      } else if (DebugCodeEnabled ()) {
+        ShellTimeout = (UINTN)PcdGet16 (PcdPlatformBootTimeOut);
+        BootShell    = TRUE;
+      }
```
Comment (gdong1) [inline PayloadPkg/OsLoader/OsLoader.c:1260]: If we boot to shell firstly, then boot to setup, is there any impact?
If it works, the setup boot option is same with other boot options, maybe we don't need add this limitation. and this behavior is also aligned with UEFI bios. (they could boot to shell firstly, then entering setup menu).

File: PayloadPkg/OsLoader/OsLoader.c (hunk 1 above)
Comment (mauricema) [reply PayloadPkg/OsLoader/OsLoader.c:697]: We might need another PCD to control the payload module support.   This is different from the Setup support.   
Payload module will enable launching from OsLoader.   Setup support will control if we can edit CFGDATA. 
I can work on it.

File: PayloadPkg/OsLoader/OsLoader.c (hunk 2 above)
Comment (mauricema) [reply PayloadPkg/OsLoader/OsLoader.c:1260]: I think we can keep this capability.  But it is platform's choice to determine if it needs to set this flag.  
For production that needs this,  I would think the restriction is required for security reason.

**Comment by mauricema** [comment]
submitted new commit to address review comments.

**Review by myguitar — APPROVED** [review]
Looks good to me

**Review by gdong1 — APPROVED** [review]
Thanks for adding this. Let's start with this firstly :)

**Pull request merged**